src/variety/lither.js

```diff
@@ -0,0 +1,327 @@
+				"bgcolor2",
+				"clear",
+				"info-line"
+			]
+		},
+		mouseinput_auto: function() {
+			var puzzle = this.puzzle;
+			if (puzzle.playmode) {
+				if (this.checkInputBGcolor()) {
+					this.inputBGcolor();
+				} else if (this.btn === "left") {
+					if (this.mousestart || this.mousemove) {
+						this.inputLine();
+					} else if (this.mouseend && this.notInputted()) {
+						this.prevPos.reset();
+						this.inputpeke();
+					}
+				} else if (this.btn === "right") {
+					if (this.mousestart || this.mousemove) {
+						this.inputpeke();
+					}
+				}
+			} else if (puzzle.editmode) {
+				if (this.mousestart) {
+					this.inputqnum();
+				}
+			}
+		},
+
+		checkInputBGcolor: function() {
+			var inputbg = this.puzzle.execConfig("bgcolor");
+			if (inputbg) {
+				if (this.mousestart) {
+					inputbg = this.getpos(0.25).oncell();
+				} else if (this.mousemove) {
+					inputbg = this.inputData >= 10;
+				} else {
+					inputbg = false;
+				}
+			}
+			return inputbg;
+		}
+	},
+
+	//---------------------------------------------------------
+	// キーボード入力系
+	KeyEvent: {
+		enablemake: true
+	},
+
+	//---------------------------------------------------------
+	// 盤面管理系
+	Cell: {
+		maxnum: 3,
+		minnum: 0,
+
+		getdir4BorderLine1: function() {
+			var adb = this.adjborder,
+				cnt = 0;
+			if (adb.top.isLine()) {
+				cnt++;
+			}
+			if (adb.bottom.isLine()) {
+				cnt++;
+			}
+			if (adb.left.isLine()) {
+				cnt++;
+			}
+			if (adb.right.isLine()) {
+				cnt++;
+			}
+			return cnt;
+		}
+	},
+
+	Board: {
+		hasborder: 2,
+		borderAsLine: true
+	},
+
+	LineGraph: {
+		enabled: true,
+		makeClist: true
+	},
+
+	//---------------------------------------------------------
+	// 画像表示系
+	Graphic: {
+		irowake: true,
+		bgcellcolor_func: "qsub2",
+		numbercolor_func: "qnum",
+		margin: 0.5,
+
+		paint: function() {
+			this.drawBGCells();
+			this.drawLines();
+			this.drawBaseMarks();
+			this.drawQuesNumbers();
+			this.drawPekes();
+			this.drawTarget();
+		},
+
+		repaintParts: function(blist) {
+			this.range.crosses = blist.crossinside();
+			this.drawBaseMarks();
+		}
+	},
+
+	//---------------------------------------------------------
+	// URLエンコード/デコード処理
+	Encode: {
+		decodePzpr: function(type) {
+			this.decode4Cell();
+		},
+		encodePzpr: function(type) {
+			this.encode4Cell();
+		},
+
+		decodeKanpen: function() {
+			this.fio.decodeCellQnum_kanpen();
+		},
+		encodeKanpen: function() {
+			this.fio.encodeCellQnum_kanpen();
+		}
+	},
+	//---------------------------------------------------------
+	FileIO: {
+		decodeData: function() {
+			if (this.filever === 1) {
+				this.decodeCellQnum();
+				this.decodeCellQsub();
+				this.decodeBorderLine();
+			} else if (this.filever === 0) {
+				this.decodeCellQnum();
+				this.decodeBorderLine();
+			}
+		},
+		encodeData: function() {
+			this.filever = 1;
+			this.encodeCellQnum();
+			this.encodeCellQsub();
+			this.encodeBorderLine();
+		},
+		kanpenOpen: function() {
```

Comment:
Same as Touch Slitherlink: Everything except `decodeData` and `encodeData` can be removed.

Reply:
Removed